### PR TITLE
dev: compatible with vanilla extract style

### PR DESCRIPTION
**Pull request by Asuka109**
## Summary

- Add example for vanilla extract css.
- Fix: externals-plugin failed to resolve falsy values.
- Fix: failed to parse request query with JSON format including # sign.
- Basic implementation of the `normal_module_factory_create_module` hook.

## Require Documentation?

- [ ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: packages/rspack/src/exports.ts [hunk 1]
```diff
@@ -71,6 +71,8 @@ export {
 	cleanupGlobalTrace as experimental_cleanupGlobalTrace
 } from "@rspack/binding";
 
+export { default as EntryOptionPlugin } from "./lib/EntryOptionPlugin";
```
Comment (ahabhgk) [inline packages/rspack/src/exports.ts:74]: EntryOptionsPlugin shouldn't be exported

Comment (Asuka109) [reply packages/rspack/src/exports.ts:74]: Is there any way to satisfy this needage?

https://github.com/vanilla-extract-css/vanilla-extract/blob/27552826716993d9923d2aa5f30513594ff3c1ee/packages/webpack-plugin/src/compiler.ts#L88-L91

Comment (Asuka109) [reply packages/rspack/src/exports.ts:74]: Should I porting it into a built-in plugin in Rust.

**Comment by Asuka109** [comment]
This PR is complete and awaiting your review~ @h-a-n-a  @ahabhgk

**Comment by Asuka109** [comment]
!canary

**Review by JSerFeng — APPROVED** [review]

**Comment by JSerFeng** [comment]
!bench

**Pull request merged**